### PR TITLE
[ENHANCEMENT] Adding tqdm to Profiler

### LINKED ISSUE #2500: Progress Bar for profiler
Some processes can take a long time, like the BasicSuiteBuilderProfiler and a progress bar can make it much more bearable.

There is a great progress bar package `tqdm` (https://github.com/tqdm/tqdm) that would be great addition to these longer process. Would this addition interest others?

### PR DESCRIPTION
**Pull request by peterdhansen**
added tqdm to BasicSuiteBuilderProfiler resolve #2500

### CODE REVIEW

**Comment by eugmandel** [comment]
@peterdhansen Thank you for creating this PR - a reviewer will take it during our triage today.

**Comment by talagluck** [comment]
Hi @peterdhansen, thanks so much - this is awesome! In the last week, we have updated the profiler we use with `great_expectations suite scaffold` from the BasicSuiteBuilderProfiler to the UserConfigurableProfiler. Would you mind adding this to that profiler as well? Happy to chat/pair on that if you have any questions.

**Comment by peterdhansen** [comment]
Added progress bar in two places for UserConfigurableProfiler

File: great_expectations/profile/user_configurable_profiler.py [hunk 1]
```diff
@@ -242,10 +244,15 @@ def _build_expectation_suite_from_semantic_types_dict(self):
...
-            )
+        with tqdm(
+            desc="Profiling Columns", total=len(self.column_info), delay=5
+        ) as pbar:
+            for column_name in self.column_info.keys():
+                pbar.set_postfix_str(f"Column={column_name}")
+                self._build_expectations_for_all_column_types(
+                    self.profile_dataset, column_name
+                )
+                pbar.update()
```
Comment (talagluck) [inline great_expectations/profile/user_configurable_profiler.py]: This is great! Just one more change - you can actually combine the call to self._build_expectations_for_all_column_types() in with the loop on lines 239-245 (this is a refactor I should have made already). Then the progress bar will apply to all of the expectations added, in addition to the non-semantic-type specific ones. The code is below (I can't comment on an unchanged line of code in GitHub, or I would suggest it directly).

```
        with tqdm(
                desc="Profiling Columns", total=len(self.column_info), delay=5
        ) as pbar:
            for column_name, column_info in self.column_info.items():
                pbar.set_postfix_str(f"Column={column_name}")
                semantic_types = column_info.get("semantic_types")
                for semantic_type in semantic_types:
                    semantic_type_fn = self.semantic_type_functions.get(semantic_type)
                    semantic_type_fn(
                        profile_dataset=self.profile_dataset, column=column_name
                    )
                    self._build_expectations_for_all_column_types(
                        self.profile_dataset, column_name
                    )
                pbar.update()
```

**Review by talagluck — COMMENTED** [review]
Looks great - thanks for your responsiveness! Just one last change to improve the functionality

**Review by talagluck — APPROVED** [review]
LGTM! Thanks for the quick turnaround

**Pull request merged**